### ISSUE TITLE
Making Accuracy support threshold = 0

### ISSUE THREAD

**Issue by cnut1648** [PyTorchLightning/metrics#416]
## 🐛 Bug
Hi, currently we can't set threshold to be 0 in Accuracy due to [the check here](https://github.com/PyTorchLightning/metrics/blob/5e29a6082d381c96d48dd942f903c24fef949343/torchmetrics/classification/accuracy.py#L208-L209). This would make this metric unable to support the output logit of the binary classification model (when there is no sigmoid).

### To Reproduce

```python
m = Accuracy(num_classes=2, average="macro", threshold=0.0)
```

### Environment

latest torchmetrics (0.4.1)
### Additional context

Related @391

**Comment by cnut1648** [comment]
Oh wait I noticed that it was already fixed in the master. Closing this issue now.
